File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/receipt/ReceiptContract.java
package org.mifos.mobilewallet.mifospay.receipt;

import org.mifos.mobilewallet.mifospay.base.BasePresenter;
import org.mifos.mobilewallet.mifospay.base.BaseView;

import okhttp3.ResponseBody;

/**
 * This is a contract class working as an Interface for UI
 * and Presenter components for receipt package.
 * @author ankur
 * @since 6-June-2018
 */

public interface ReceiptContract {

    /**
     * Defines all the functions in UI component.
     */
    interface ReceiptView extends BaseView<ReceiptPresenter> {

        void showSnackbar(String message);

        void writeReceipt(ResponseBody responseBody, String filename);

        void hideProgressDialog();
    }

    /**
     * Defines all the functions in Presenter component.
     */
    interface ReceiptPresenter extends BasePresenter {

        void fetchReceipt(String transactionId);
    }
}

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/receipt/ReceiptContract.kt
package org.mifos.mobilewallet.mifospay.receipt

import okhttp3.ResponseBody
import com.mifos.mobilewallet.model.entity.accounts.savings.TransferDetail
import com.mifos.mobilewallet.model.domain.Transaction
import org.mifos.mobilewallet.mifospay.base.BasePresenter
import org.mifos.mobilewallet.mifospay.base.BaseView

/**
 * Created by ankur on 06/June/2018
 */
interface ReceiptContract {
    interface ReceiptView : BaseView<ReceiptPresenter?> {
        fun showSnackbar(message: String?)
        fun writeReceiptToPDF(responseBody: ResponseBody?, filename: String?)
        fun hideProgressDialog()
        fun showTransactionDetail(transaction: Transaction?)
        fun showTransferDetail(transferDetail: TransferDetail?)
        fun openPassCodeActivity()
    }

    interface ReceiptPresenter : BasePresenter {
        fun downloadReceipt(transactionId: String?)
        fun fetchTransaction(transactionId: Long)
    }
}